contracts/l2/curation/L2Curation.sol

```diff
@@ -296,17 +284,22 @@ contract L2Curation is CurationV2Storage, GraphUpgradeable, IL2Curation {
         uint256 _tokensOutMin
     ) external override notPartialPaused returns (uint256) {
         address curator = msg.sender;
-
+        uint256 previousCuratorSignal = getCuratorSignal(curator, _subgraphDeploymentID);
         // Validations
         require(_signalIn != 0, "Cannot burn zero signal");
-        require(
-            getCuratorSignal(curator, _subgraphDeploymentID) >= _signalIn,
-            "Cannot burn more signal than you own"
-        );
+        require(previousCuratorSignal >= _signalIn, "Cannot burn more signal than you own");
 
         // Get the amount of tokens to refund based on returned signal
         uint256 tokensOut = signalToTokens(_subgraphDeploymentID, _signalIn);
-
+        if (previousCuratorSignal > _signalIn) {
```

Comment:
pedantic nit, this could be `previousCuratorSignal != _signalIn` and save a little gas. Though not sure how you feel about code readability.

Reply:
Good point, but yeah for now I'll leave it like this for readability